bin/plugin/commands/test-plugins.js

```diff
@@ -0,0 +1,440 @@
+			);
+		}
+
+		log( command.stdout.replace( '\n', '' ) );
+
+		if ( 1 === command.status ) {
+			log(
+				formats.error(
+					`One or more tests failed for plugin ${ plugin }`
+				)
+			);
+
+			// Return with exit code 1 to trigger a failure in the test pipeline.
+			process.exit( 1 );
+		}
+	} );
+
+	// Start winding down.
+	log( `Stopping wp-env...` );
+
+	// Stop wp-env.
+	execSync( `wp-env stop`, ( err, output ) => {
+		// once the command has completed, the callback function is called.
+		if ( err ) {
+			log( formats.error( `${ err }` ) );
+			return;
+		}
+		// log the output received from the command.
+		log( output );
+	} );
+}
+
+/**
+ * Runs standalone plugin tests in single or multisite environments.
+ *
+ * @param {WPTestPluginsSettings} settings Plugin test settings.
+ */
+function doRunStandalonePluginTests( settings ) {
+	// Check if the siteType arg is one of single or multi.
+	if (
+		'single' !== settings.siteType &&
+		'multi' !== settings.siteType
+	) {
+		log(
+			formats.error(
+				`--sitetype must be one of "single" or "multi".`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// If the base .wp-env.json file for testing plugins is missing, abort.
+	if ( ! fs.pathExistsSync( settings.wpEnvFile ) ) {
+		log(
+			formats.error(
+				`WP Env config file "${ settings.wpEnvFile }" not detected in root of project.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Buffer built plugins array.
+	let builtPlugins = [];
+
+	// Buffer contents of plugins JSON file.
+	let pluginsJsonFileContent = '';
+
+	try {
+		pluginsJsonFileContent = fs.readFileSync( settings.pluginsJsonFile, 'utf-8' );
+	} catch ( e ) {
+		log(
+			formats.error( `Error reading file at "${ settings.pluginsJsonFile }". ${ e }` )
+		);
+	}
+
+	// Validate that the plugins JSON file contains content before proceeding.
+	if (
+		'' === pluginsJsonFileContent ||
+		! pluginsJsonFileContent
+	) {
+		log(
+			formats.error(
+				`Contents of file at "${ settings.pluginsJsonFile }" could not be read, or are empty.`
+			)
+		);
+	}
+
+	const pluginsJsonFileContentAsJson = JSON.parse( pluginsJsonFileContent );
+
+	// Check for valid and not empty object resulting from plugins JSON file parse.
+	if (
+		'object' !== typeof pluginsJsonFileContentAsJson ||
+		0 === Object.keys( pluginsJsonFileContentAsJson ).length
+	) {
+		log(
+			formats.error(
+				`File at "settings.pluginsJsonFile" parsed, but detected empty/non valid JSON object.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Create an array of plugins from entries in plugins JSON file.
+	builtPlugins = Object.keys( pluginsJsonFileContentAsJson )
+		.filter( ( item ) => {
+			if ( ! fs.pathExistsSync( `${ settings.builtPluginsDir }${ pluginsJsonFileContentAsJson[ item ].slug }` ) ) {
```

Comment:
Do we need to add any handling here in case the `builtPluginsDir` value does not end in a slash?

Reply:
@joemcgill  I do not think this is absolutely necessary, as this is not a dynamically defined value, we hard define it as part of the settings object we pass to `doRunStandalonePluginTests()`. We can potentially handle this in a future PR if the need arises?